### PR TITLE
[[ Hooks ]] Add ability to hook into certain aspects of engine functionality.

### PR DESCRIPTION
**Pull request by runrevmark**
The MCHook API is a simple way for parts of the engine to be extended. Currently extra tables of 'global handlers' and 'native controls' can be added.

As it stands this integration still needs to be linked into the main engine - it isn't a clean enough API for public use.

### CODE REVIEW

File: config.sh [hunk 1]
```diff
@@ -266,7 +266,7 @@ WIN_PERL=${WIN_PERL:-"C:/perl/bin/perl.exe"}
 # Android default settings and tools
 if test "${OS}" = "android" ; then
     ANDROID_NDK_VERSION=${ANDROID_NDK_VERSION:-r10d}
-    ANDROID_PLATFORM=${ANDROID_PLATFORM:-android-8}
+    ANDROID_PLATFORM=${ANDROID_PLATFORM:-android-10}
```
Comment (livecodefraser) [inline config.sh:269]: Although we don't have very many (any?) users of Android 2.2, a release note would be useful to say that we now require 2.3.3.

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay e45d77d

Marking as reviewed because it looks not-totally-unreasonable (release note excepted) and it could do with a compile test ;)

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok e45d77dffb2f21e3d639dbb5587f41f1ca79f0cb

File: engine/src/exec-nativecontrol.cpp [hunk 2]
```diff
@@ -613,7 +613,7 @@ void MCNativeControlExecCreateControl(MCExecContext& ctxt, MCStringRef p_type_na
     }
     
     MCNativeControlType t_type;
-    if (!MCNativeControl::LookupType(p_type_name, t_type))
+    if (!MCLookupNativeControlType(p_type_name, (intenum_t&)t_type))
```
Comment (peter-b) [inline engine/src/exec-nativecontrol.cpp:616]: Maybe these would be better with `reinterpret_cast`?

**Comment by peter-b** [comment]
In general, this changeset has a very unfortunate lack of comments that describe what the various newly-introduced types and functions and methods and stuff do and why they do it that way. And cryptic commit messages. :sob:

**Comment by livecode-vulcan** [comment]
😎 test success e45d77dffb2f21e3d639dbb5587f41f1ca79f0cb
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/740)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/800)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/690)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/754)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/685)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/677)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 2116c42

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 2116c4249ed71e8514d288c508d83c8788679da1

**Comment by livecode-vulcan** [comment]
😎 test success 2116c4249ed71e8514d288c508d83c8788679da1
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/7)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/7)
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/7)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/7)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/7)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/7)

**Pull request merged**